### PR TITLE
completed 3-get-articles-by-id

### PR DESCRIPTION
**Pull request by jwiszowaty**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -36,11 +37,48 @@ describe('GET /api', () => {
...
+        .then(({body}) => {
+            expect(body.article_request).toHaveLength(1)
+            expect(Object.keys(body.article_request[0]).includes('author')).toBe(true)
+            expect(Object.keys(body.article_request[0]).includes('title')).toBe(true)
+            expect(Object.keys(body.article_request[0]).includes('article_id')).toBe(true)
+            expect(Object.keys(body.article_request[0]).includes('body')).toBe(true)
+            expect(Object.keys(body.article_request[0]).includes('topic')).toBe(true)
+            expect(Object.keys(body.article_request[0]).includes('created_at')).toBe(true)
+            expect(Object.keys(body.article_request[0]).includes('votes')).toBe(true)
+            expect(Object.keys(body.article_request[0]).includes('article_img_url')).toBe(true)
```
Comment (aloush-dev) [inline __tests__/app.test.js]: Can we think of a better way to test the article we are getting back?
Currently we could be getting any article back and not the one we requested.
Take a look at the Jest matchers, ObjectContaining and ToMatchObject.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -36,11 +37,48 @@ describe('GET /api', () => {
...
+        })
+    })
+    it('returns status 404 when id does not correspond to an existing article_id', () => {
+        return request(app)
+        .get('/api/articles/17')
+        .expect(404)
+        .then((result) => {
+            const {msg} = JSON.parse(result.error.text)
```
Comment (aloush-dev) [inline __tests__/app.test.js]: we shouldn't need to parse things in the test file to gain access to their contents.

**Review by aloush-dev — APPROVED** [review]
Nice job, great seperation of the error handlers.
I think with the way your error handlers are laid out you will never hit you no such path yet error handler as your serverError handler will be the last one in the chain.
Please see comment regarding your test and refactor.

File: __tests__/app.test.js (hunk 2 above)
Comment (jwiszowaty) [reply __tests__/app.test.js:71]: result.error.text is always a string, I couldn't think of another way to access the value of the msg

**Pull request merged**